### PR TITLE
support no apllication of chat template for instructions

### PR DESCRIPTION
**Pull request by Ssukriti**
This change exposes a flag `apply_chat_instruction_template` to allow users to control if chat template is applied or not by default to prompt/completion data format supported by SFT Trainer.

if flag is set to False, the prompt and completion are just concatenated and masking is applied using length of prompt .

### CODE REVIEW

File: trl/trainer/sft_trainer.py [hunk 1]
```diff
@@ -250,7 +251,7 @@ def make_inputs_require_grad(module, input, output):
         if formatting_func is None and dataset_text_field is None:
             # check if dataset has ChatML format or instruction format and is supported
             # if not stays #None
-            formatting_func = get_formatting_func_from_dataset(train_dataset, tokenizer)
+            formatting_func = get_formatting_func_from_dataset(train_dataset, tokenizer, apply_chat_instruction_template)
 
         requires_input_output_keys = False
```
Comment (alex-jw-brooks) [inline trl/trainer/sft_trainer.py]: This variable name should probably be changed - it doesn't make sense if the keys are prompt / completion

Comment (Ssukriti) [reply trl/trainer/sft_trainer.py]: ya I changed it below and left that one place, thanks for catching it
